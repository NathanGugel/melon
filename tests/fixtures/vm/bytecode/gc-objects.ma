@function pushObjects 0 3
    .data
        key "key"
        value "This is a new and long value"

    .text
        PUSHI 100
        STORELOC 0
        PUSHI 0
        STORELOC 1

    :loop
        PUSHLOC 1
        PUSHLOC 0
        LT
        TESTTRUE 1
        JMP exit

        NEWOBJ
        STORELOC 2

        PUSHLOC 2
        PUSHK key
        PUSHK value
        SETOBJ

        PUSHLOC 2

        PUSHLOC 1
        PUSHI 1
        ADD
        STORELOC 1
        JMP loop
        
    :exit
        RETURN 0
@end

.text
    NEWCL pushObjects
    CALL 0 1
    GC
    DUMPINFO 2
