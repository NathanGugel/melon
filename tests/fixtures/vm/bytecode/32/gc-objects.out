GC Allocated Objs: 512
